compiler/rustc_middle/src/query/mod.rs

```diff
@@ -830,7 +830,8 @@ rustc_queries! {
         desc { |tcx| "checking privacy in {}", describe_as_module(key, tcx) }
     }
 
-    query check_liveness(key: DefId) {
+    query check_liveness(key: DefId) -> Option<FxIndexSet<Span>> {
```

Comment:
Spans are not a very robust source of information. Do we have a better, more "semantic" source of information?

Reply:
I tried but couldn't find one. Rather, there is a lot on liveness side working on HIR, but they are missing on borrowck side working on MIR, which needs to use this information.